presto-druid/src/test/java/io/prestosql/plugin/druid/TestingDruidServer.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.druid;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Closer;
+import com.google.common.io.Resources;
+import io.prestosql.testing.assertions.Assert;
+import io.prestosql.tests.product.launcher.testcontainers.SelectedPortWaitStrategy;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.shaded.okhttp3.OkHttpClient;
+import org.testcontainers.shaded.okhttp3.Request;
+import org.testcontainers.shaded.okhttp3.RequestBody;
+import org.testcontainers.shaded.okhttp3.Response;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Map;
+
+import static com.google.common.io.Resources.getResource;
+import static java.lang.String.format;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+public class TestingDruidServer
+        implements Closeable
+{
+    private final String hostWorkingDirectory;
+    private final GenericContainer broker;
+    private final GenericContainer coordinator;
+    private final GenericContainer historical;
+    private final GenericContainer middleManager;
+    private final GenericContainer zookeeper;
+    private final OkHttpClient httpClient;
+
+    private static final int DRUID_COORDINATOR_PORT = 8081;
+    private static final int DRUID_BROKER_PORT = 8082;
+    private static final int DRUID_HISTORICAL_PORT = 8083;
+    private static final int DRUID_MIDDLE_MANAGER_PORT = 8091;
+
+    private static final String DRUID_DOCKER_IMAGE = "apache/druid:0.18.0";
+
+    public TestingDruidServer()
+    {
+        try {
+            this.hostWorkingDirectory = TestingDruidServer.class.getClassLoader().getResource("druid-test-storage/").getPath();
+            File f = new File(hostWorkingDirectory);
+            cleanDirectory(f);
+            // Enable read/write/exec access for the services running in containers
+            f.setWritable(true, false);
+            f.setReadable(true, false);
+            f.setExecutable(true, false);
```

Comment:
```
// Cannot use Files.createTempDirectory() because on Mac by default it uses /var/folders/ which is not visible to Docker for Mac
Path hostWorkingDirectory = Files.createDirectory(Paths.get("/tmp/druid-workdir-" + randomUUID().toString()));
```

and remove `druid-test-storage` from resources

Reply:
Unfortunately, that doesn't work. The GenericContainer library expects the shared directory to be a resource in the class path otherwise we get a `directory doesn't exist: access denied` error. See https://www.testcontainers.org/features/files/

Reply:
That's interesting, because we use this pattern elsewhere. Let me look into this.

Reply:
I did this:

```
// Cannot use Files.createTempDirectory() because on Mac by default it uses /var/folders/ which is not visible to Docker for Mac
hostWorkingDirectory = Files.createDirectory(Paths.get("/tmp/docker-files-" + randomUUID().toString()));
```

then

```
.withFileSystemBind(hostWorkingDirectory.toString(), "/opt/druid/var", BindMode.READ_WRITE)
```

it was correctly bound, but something didn't work quite well (the data import).
Please try to complete this this way. If you cannot, add a TODO -- using classpath resource as R/W dir is not a good idea.

Reply:
Right, I have attempted this as well but unfortunately, the middle manager service running data ingestion runs into issues since it is not able to create directories for storing logs, intermediate segments, etc. I will add a TODO.

Reply:
Pushed a commit that attempts to use /tmp/docker-files-UUID as host working directory. The test passed locally on my mac, would be interesting to see if it passes on CI. I have currently removed the assertion to check if `file.delete()` was successful.

Reply:
> I have currently removed the assertion to check if file.delete() was successful.

Since we know it's not, let's restore the Guava-way, but have it commented out with a TODO.

Reply:
FYI, it worked for me locally too. @samarthjain thanks for nailing this down.

Reply:
Looks like I forgot removing one assertion in the last commit. Either way, looks like the tests did pass on CI as well.

I think it still makes sense to instead use the `deleteRecursively` method I had with the assert removed. Since things work fine locally, for anyone running this test, it should clean things up. WDYT?

Reply:
The bottom line is -- if we want to provide "best effort" delete, we should call `MoreFiles.deleteRecursively` and suppress exception.
There is no point in reimplementing the logic just for that.

i checked the current code (`Attempt using /tmp/docker-files-UUID as host working directory`, rebased on `master`), and replaced `deleteRecursively` with
```
closer.register(() -> MoreFiles.deleteRecursively(Paths.get(hostWorkingDirectory), RecursiveDeleteOption.ALLOW_INSECURE));
```

and... did not get the failure back. Very confusing. Can this be in some way non-deterministic?

Reply:
Fair point on suppressing the exception for now. 
Locally, MoreFiles.deleteRecursively() works fine. It is on CI that it fails.